source/common/upstream/outlier_detection_impl.cc

```diff
@@ -248,6 +248,13 @@ void DetectorImpl::onConsecutive5xxWorker(HostSharedPtr host) {
     return;
   }
 
+  // There are two outcomes here. The ejection will be enforced,
```

Comment:
OK I see. I just meant move this code to after the ejectHost() call (so end of function). It will happen whether it gets ejected or not but in the case it is ejected there will be no race. Please reflow the comment and you have a small typo below "host keeps been".